stable/velero/README.md

```diff
@@ -1,117 +1,100 @@
+
+#### Option 1) CLI commands
+
+Specify the necessary values using the --set key=value[,key=value] argument to helm install. For example,
+
+```bash
+helm install --namespace <YOUR NAMESPACE> \
+--set configuration.provider=<PROVIDER NAME> \
+--set-file credentials.secretContents.cloud=<FULL PATH TO FILE> \
+--set configuration.backupStorageLocation.name=<PROVIDER NAME> \
+--set configuration.backupStorageLocation.bucket=<BUCKET NAME> \
+--set configuration.backupStorageLocation.config.region=<REGION> \
+--set configuration.volumeSnapshotLocation.name=<PROVIDER NAME> \
+--set configuration.volumeSnapshotLocation.config.region=<REGION> \
+--set image.repository=velero/velero \
+--set image.tag=v1.2.0 \
+--set image.pullPolicy=IfNotPresent \
+--set initContainers[0].name=velero-plugin-for-aws \
+--set initContainers[0].image=velero/velero-plugin-for-aws:v1.0.0 \
+--set initContainers[0].volumeMounts[0].mountPath=/target \
+--set initContainers[0].volumeMounts[0].name=plugins \
+stable/velero
+```
+
+#### Option 2) YAML file
+
+Add/update the necessary values by changing the values.yaml from this repository, then running:
+
+```bash
+helm install --namespace <NAMESPACE> -f values.yaml stable/velero
+```
+
+#### Upgrade the configuration
+
+If a value needs to be added or changed, you may do so with the `upgrade` command. An example:
+
+```bash
+helm upgrade <RELEASE NAME> --set initContainers.backupStorageLocation.name=aws,initContainers.volumeSnapshotLocation.name=aws stable/velero
+```
+
+## Upgrading
+
+### Upgrading to v1.1.0
 
 The [instructions found here](https://velero.io/docs/v1.1.0/upgrade-to-1.1/) will assist you in upgrading from version v1.0.0 to v1.1.0
 
-## Upgrading to v1.0.0
+### Upgrading to v1.0.0
 
 As of v1.0.0, Velero is no longer backwards-compatible with Heptio Ark.
 
 The [instructions found here](https://velero.io/docs/v1.0.0/upgrade-to-1.0/) will assist you in upgrading from version v0.11.0 to v1.0.0
 
-## Upgrading to v0.11.0
+### Upgrading to v0.11.0
 
 As of v0.11.0, Heptio Ark has become Velero.
 
 The [instructions found here](https://velero.io/docs/v0.11.0/migrating-to-velero/) will assist you in upgrading from Ark to Velero
 
-## Prerequisites
-
-### Secret for cloud provider credentials
-Velero server needs an IAM service account in order to run, if you don't have it you must create it.
-Please follow the official documentation: https://velero.io/docs/v1.0.0/install-overview/
+## Uninstall Velero
 
-Don't forget the step to create the secret
-```
-kubectl create secret generic cloud-credentials --namespace <VELERO_NAMESPACE> --from-file cloud=credentials-velero
-```
+Note: when you uninstall the Velero server, all backups remain untouched.
 
-### Configuration
```

Comment:
Is this information added elsewhere? There's the CLI command example, but it lacks the description/required fields from this table, which I find useful.

Reply:
We had talked about this in the past in the context of the `velero install` command. What we had decided was to have the CLI itself be the source of documentation, instead of trying to keep such documentation up to date in multiple places. At least this is my recollection, and also doesn't mean we can't change it.

Reply:
Yeah, that's true of `velero install`, though the Helm chart is a bit different. For example, all of these options can (an normally, should) be in a values.yaml file versus just being on the command line. Also, when they're on the command line, they're not their own flags, but rather key/value pairs given to `set`.

There's also options like `credentials.existingSecret` that aren't in `velero install` that have some interactions with other values. That's not immediately clear from just the command line instructions here.

Finally, there's also [Helm standards](https://helm.sh/docs/chart_best_practices/#document-values-yaml) that mention documenting what's possible in a `values.yaml` file.

I think I'd propose having both the CLI example that was added as well as the table here.

Reply:
Gotcha. But these values are all documented in `values.yaml`. Do they need to be duplicated in the readme too?

Reply:
No, I don't think they need to duplicated - in the `values.yaml` is fine. I'll mark this as resolved.

Reply:
I think the convention is to actually have them documented. Personally I find dot notation way better searchable than YAML notation. Example:

`configuration.backupStorageLocation.config.storageAccount` 
vs
```
configuration:
  # a lot of documentation
  # a lot of documentation
  # a lot of documentation
  # a lot of documentation
  # a lot of documentation
  backupStorageLocation:
    # a lot of documentation
    # a lot of documentation
    # a lot of documentation
    # a lot of documentation
    # a lot of documentation
    config:
      # a lot of documentation
      # a lot of documentation
      # a lot of documentation
      # a lot of documentation
      # a lot of documentation
      storageAccount

Reply:
That's a good point. While it's well taken, I think in order to get Helm users onto v1.2, we should defer this discussion to a separate PR and focus this one on any changes necessary to get 1.2 installable at this stage.